tools/common_lib/src/gemm.h

```diff
@@ -1285,9 +1285,19 @@ class GemmUmdD3d12Dispatcher : public GemmBaseDispatcher
             return dnnl::memory{};
         }();
 
-        dnnl::memory scratchpad_memory = has_scratchpad_tensor() ?
-            create_dnnl_memory(scratchpad_memory_desc_.value(), umd_scratchpad_memory_) :
-            dnnl::memory{};
+        // dnnl::memory scratchpad_memory = has_scratchpad_tensor() ?
+        //     create_dnnl_memory(scratchpad_memory_desc_.value(), umd_scratchpad_memory_) :
+        //     dnnl::memory{};
+
+        std::optional<dnnl::memory> scratchpad_memory;
```

Comment:
Wasn't commented code enough to create scratchpad dnnl::memory object?

Reply:
It is enough to create object, but later on, we use `scratchpad_memory_desc_.value()` which only works for std::optional. That is the code piece I captured from convolution code. To keep it the same style, I commented the current code.